### PR TITLE
Allow (mobile) plugins to define telemetry logs.

### PR DESCRIPTION
**Pull request by dogmaphobic**

### CODE REVIEW

File: src/QGCApplication.cc [hunk 1]
```diff
@@ -503,31 +503,39 @@ void QGCApplication::criticalMessageBoxOnMainThread(const QString& title, const
...
-        QString dtFormat("yyyy-MM-dd hh-mm-ss");
-
-        int tryIndex = 1;
-        QString saveFileName = nameFormat.arg(QDateTime::currentDateTime().toString(dtFormat)).arg("");
-        while (saveDir.exists(saveFileName)) {
-            saveFileName = nameFormat.arg(QDateTime::currentDateTime().toString(dtFormat)).arg(QStringLiteral(".%1").arg(tryIndex++));
-        }
-        QString saveFilePath = saveDir.absoluteFilePath(saveFileName);
+    // Are we supposed to log?
+    if(toolbox()->settingsManager()->appSettings()->telemetrySave()->rawValue().toBool()) {
```
Comment (DonLakeFlyer) [inline src/QGCApplication.cc]: You shouldn't need this. You should never get here if that isn't set.

Comment (DonLakeFlyer) [inline src/QGCApplication.cc]: The only case would be if you crash and there is a leftover log file. But if that happens then save was on at the time you crashed. Hence you would want to save that anyway.

Comment (DonLakeFlyer) [inline src/QGCApplication.cc]: Ah, I hate the stupid review thing...

**Comment by dogmaphobic** [comment]
I don't know what I did but your notes, along with my answers disappeared. I guess I see why you hate these code reviews :)
Either way, yes. I had added that before understanding the *Always Write To Temp* deal. I've removed it.

**Pull request merged**